Update glossary.mdx (#11)
federated-analytics/docs: website/pages/glossary.mdx

# Glossary of Terms

This a glossary of terms which relate to the techonology stack and general research environment in which this technology stack will be used. 

# Actors
- **User Manager:** This is the manager of the Keycloak server. They deal with user accounts for all services.
- **TRE Manager:** This is the manager of a TRE server. They deal with project and user approvals, as well as connecting to the Egress server.
- **Submission Manager:** This is the manager of the submission layer. They create projects and people, and assign people to projects.
- **Egress Manager:** This is the manager of the egress layer. They deal with approving data release from TREs.
- **Researcher:** This is the researcher using the service. They will be registered to use the submission layer, and they may be associated with one or more projects. They will be submitting tasks to be run in the TREs.

# TRE-FX Services
- **Egress Layer:** In TRE-FX context, the egress layer serves as a temporary holding space for data returned from the Trusted Research Environment which needs to be authorised for release before being passed to the Submission Layer.

> - **HUTCH:** This is an open-source tool designed to facilitate federated activities - such as analytics, data discovery and machine learning within secure environments.

Reviewer: ```suggestion
- **HUTCH:** This is an open-source toolset designed to facilitate federated activities - such as analytics, data discovery and machine learning within secure environments.
```

> - **Submission Layer:** In TRE-FX contex, the submission layer serves as the intial point of contact for incoming analytical tasks, responsible for receiving user-submitted tasks, performing preliminary validation and queuing the tasks for further processing.

Reviewer: ```suggestion
- **Submission Layer:** In TRE-FX context, the submission layer serves as the initial point of contact for incoming analytical tasks, responsible for receiving user-submitted tasks, performing preliminary validation and queuing the tasks for further processing.
```

> - **Submission Layer:** In TRE-FX context, the submission layer serves as the initial point of contact for incoming analytical tasks, responsible for receiving user-submitted tasks, performing preliminary validation and queuing the tasks for further processing.

Reviewer 2: I've adjusted the wording a bit - the original does imply data being returned to the submission layer, but I've tried to make this a bit more explicit.

> - **TRE Controller Layer:** This is the outer layer of the TRE, responsible for task scheduling, data coordination and messaging between the different layers of the TRE.

Reviewer: ```suggestion
- **TRE Agent Layer:** This is the outer layer of the TRE, responsible for task scheduling, data coordination and messaging between the different layers of the TRE.
```

- **TRE-UI:** This is the user interface that offers an integrated view of ongoing tasks, logs and system metrics.
- **TRINO:** This is a distributed SQL query engine designed to query large data sets distributed over one or more heterogeneous data sources.

> - **Wizard UI:** A user interface that guides users through a series of steps to complete, allowing the user to break down complex processes into small steps to achieve a goal.

Reviewer 2: I think we need someone involved in the Wizard UI to define this. @LockhartJones or @beforan - could you add to this?

# Trusted Research Environment / Health Research Terms
- **Data Anonymisation:** This is the removal or protection of any identifiable personal information from datasets.

> - **Data Provenance:** This refers to the history and origin of data.

Reviewer 2: Could we go with:

This includes the origin of data, how it was created and processed, who was involved in this, and when it was made and modified.

> - **Data Provenance:** This refers to the history and origin of data.

Reviewer 2: I've added my suggested text.

- **Dataset:** A collection of data, often structured in columns and rows (tabular form). Datasets can differ in sizes from a few records to billon of data points.
- **Egress:** This refers to data leaving an environment or a network system.
- **FAIR:** Findable, Acceptable, Interoperable and Re-usable. This is a framework used in the research space.
- **Federated Activities (Trusted Research Environment):** This a network of TRE institutions that collaborate among themselves without compromising data privacy and security.
- **Federated Query:** This involves getting data from multiple databases as if it were one single database.
- **Ingress:** This refers to data entering a system or a network. 
- **Metadata:** Important information associated with the data. It details the origin, version, authors, source data, and other information about the data set.

> - **OMOP:** Observational Medical Outcomes Partnership

Reviewer 2: I've added a CDM entry, and added some descriptive text here.

- **Research Data:** This refers to information that has been collected, analysed and generated during the process of a research project or scientific investigation. Research data can be collected via processes such as interviews, survey results, clinical trial outcomes, or chemical analysis.
- **RO Crate:** A transport format to bundle metadata for submission along with required input parameters and everything implementation needs to know in order to perform the requested analysis
It is a methodology of packaging research data, with it's associated metedata and it's components and files."
- **SDE:** Secure Data Environment- a platform where sensitive health and care data is stored securely which can be used for research and analysis.
- **Task:** A request submitted by a submission layer User as part of a project, for some analysis or computation within one or more Trusted Research Environments.
- **TRE (Trusted Research Environment):** This is a secure digital environment that has valuable de-identified health and care data that can only be accessed by approved researchers.

> # Technology

Reviewer 3: I've gone through some of the technical terms and suggested what I'd put there.

> - **API (Application Programming Interface):** This comprises of guidelines, and protocols that enable software applications to interact with each other by specifying the techniques and data structures for requesting and sharing information.

Reviewer 3: API: A set of guidelines and protocols that enable software applications to interact with each other by specifying the techniques and data structures for requesting and sharing information.

> - **Application Stack:** This is a collection of software technologies and tools that collaborate to develop and operate an application efficiently and effectively.

Reviewer 3: Application Stack: A collection of software technologies and tools that work together to develop and operate an application.

- **Database:** This is a collection of data that is organised, stored and managed electronically.

> - **Database Credentials:** These are required pieces of information that allows an individual or an application to access a database. This usually comprises of a Username( the identifier for the user of the application), Password ( A secret key known to the user/users associated with the username to authenticate access).

Reviewer 3: Do we need to define username and password? I think even to a non-technical audience this is straightforward.

- **Docker Compose:** This is a tool for defining and running multi container applications.It allows user to launch an entire application stack using the configuration YAML file.
- **GUI (Graphical User Interface):** This is a user interface where the interaction is with the device or computer is done by the use of visual elements like icons, windows, menus and buttons.

> - **Kubernetes:** Abbreviated as K8. This is an open-source platform that helps manage and automate the deployment, scaling and operations of containers.

Reviewer 3: Kubernetes (K8s):  An open-source platform that automates the deployment, scaling, and management of containerised applications.

> - **MinIO:** "This is a software application that provides storage space that allows for the storage of large amounts of data.

Reviewer 3: We need to state that it's _object storage_. 
> A high-performance, software-defined object storage system compatible with Amazon S3.
In a TRE-FX context, it is used as an intermediary storage layer for holding the Five Safes RO-Crates during ingress and egress.

Reviewer 2: Thanks for this - I'll adapt the text.

I don't think we need to give the TRE-FX context, as this might change (e.g. we might end up pushing to another object store on egress from the service).

In a TRE-FX context, it is used as an intermediary storage layer for holding the Five Safes RO-Crates during ingress and egress.

> - **Open source:** This is a software with a code that can be assessed, maintained and improved through open collaboration.

Reviewer 3: I'd mention licenses perhaps:
> Software with source code that can be accessed, modified, and improved through open collaboration. Open-source software is typically released under licenses that define usage, distribution, and modification rights.

> - **PostgressSQL 14:** A database which uses tables (row and columns to store data).This is used for storing essential data. i.e. Which users have been authorised to run analyses against which projects, task statuses.

Reviewer 3: Typo: "PostgressSQL" → "PostgreSQL."
> PostgreSQL: An open-source relational database management system (RDBMS) that uses tables (rows and columns) to store data. In this context, it stores essential information, such as which users are authorised to run analyses against specific projects and task statuses.

- **Repository:** A storage location where project assets such as requirements, code, files are managed and stored. It assists in maintaining a structured development process.

> - **REST API:** Representational State Transfer Application Programming Interface.

Reviewer 3: REST API: An architectural style for designing networked applications. REST APIs use HTTP requests to perform standard operations such as creating, reading, updating, or deleting (CRUD) resources. An example is a service requesting data from another web service.
